<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1329</width>
    <height>920</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1329</width>
    <height>920</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="glView" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>5</y>
      <width>910</width>
      <height>910</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="open_file">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>110</y>
      <width>161</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	
	font: 13pt &quot;Apple Braille&quot;;
     border-radius: 8px;
     background-color:  #008000	;
     min-width: 80px;
 }

QPushButton:pressed {
     background-color: #00ff00;
 }</string>
    </property>
    <property name="text">
     <string>OPEN FILE</string>
    </property>
   </widget>
   <widget class="QLabel" name="filename_label">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>5</y>
      <width>321</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Filename:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_x">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>240</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_y">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>240</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_z">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>240</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>210</y>
      <width>261</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Move model:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_z_2">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>350</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_y_2">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>350</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>320</y>
      <width>261</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate model:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_x_2">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>350</y>
      <width>70</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>430</y>
      <width>58</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Scaling:</string>
    </property>
   </widget>
   <widget class="QLabel" name="vertex_count_label">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>40</y>
      <width>321</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex count:</string>
    </property>
   </widget>
   <widget class="QLabel" name="facets_count_label">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>70</y>
      <width>321</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Facets count:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox">
    <property name="geometry">
     <rect>
      <x>932</x>
      <y>170</y>
      <width>191</width>
      <height>32</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Parallel projection</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Central projection</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="green_value">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>610</y>
      <width>25</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="blue_value">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>660</y>
      <width>25</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QSlider" name="blue_index">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>655</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="save_gif">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>150</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	
	font: 13pt &quot;Apple Braille&quot;;
     border-radius: 8px;
     background-color:  #008000;
     min-width: 80px;
 }

QPushButton:pressed {
     background-color: #00ff00;
 }</string>
    </property>
    <property name="text">
     <string>GIF</string>
    </property>
   </widget>
   <widget class="QSlider" name="line_size">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>820</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="line_size_value">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>825</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="point_size_value">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>765</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_image">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>110</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	
	font: 13pt &quot;Apple Braille&quot;;
     border-radius: 8px;
     background-color:  #008000;
     min-width: 80px;
 }

QPushButton:pressed {
     background-color: #00ff00;
 }</string>
    </property>
    <property name="text">
     <string>IMAGE</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>610</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: green;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>green</string>
    </property>
   </widget>
   <widget class="QSlider" name="point_size">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>760</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>810</y>
      <width>73</width>
      <height>58</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="1" column="0">
      <widget class="QRadioButton" name="dashed_line">
       <property name="styleSheet">
        <string notr="true">QRadioButton{
color: white;
	font: 13pt &quot;Apple Braille&quot;;
}</string>
       </property>
       <property name="text">
        <string>Dashed</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QRadioButton" name="solid_line">
       <property name="styleSheet">
        <string notr="true">QRadioButton{
color: white;
	font: 13pt &quot;Apple Braille&quot;;
}</string>
       </property>
       <property name="text">
        <string>Solid</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QRadioButton" name="exist_line">
       <property name="styleSheet">
        <string notr="true">QRadioButton{
color: white;
	font: 13pt &quot;Apple Braille&quot;;
}</string>
       </property>
       <property name="text">
        <string>exist</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>790</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>Line size</string>
    </property>
   </widget>
   <widget class="QSlider" name="green_index">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>605</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>730</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>Point size</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>740</y>
      <width>73</width>
      <height>58</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="0">
      <widget class="QRadioButton" name="point_type_circle">
       <property name="styleSheet">
        <string notr="true">QRadioButton{
color: white;
	font: 13pt &quot;Apple Braille&quot;;
}</string>
       </property>
       <property name="text">
        <string>Circle</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QRadioButton" name="point_type_cube">
       <property name="styleSheet">
        <string notr="true">QRadioButton{
color: white;
	font: 13pt &quot;Apple Braille&quot;;
}</string>
       </property>
       <property name="text">
        <string>Cube</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QRadioButton" name="point_type_exist">
       <property name="styleSheet">
        <string notr="true">QRadioButton{
color: white;
	font: 13pt &quot;Apple Braille&quot;;
}</string>
       </property>
       <property name="text">
        <string>exist</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QSlider" name="red_index">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>555</y>
      <width>160</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>560</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: red;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>red</string>
    </property>
   </widget>
   <widget class="QLabel" name="red_value">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>560</y>
      <width>25</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>660</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>blue</string>
    </property>
   </widget>
   <widget class="QComboBox" name="change_mode">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>510</y>
      <width>231</width>
      <height>32</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QComboBox{
	font: 13pt &quot;Apple Braille&quot;;
}</string>
    </property>
    <item>
     <property name="text">
      <string>Lines</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Points</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Background</string>
     </property>
    </item>
   </widget>
   <widget class="QDoubleSpinBox" name="scale">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>430</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="decimals">
     <number>5</number>
    </property>
    <property name="minimum">
     <double>0.000100000000000</double>
    </property>
    <property name="singleStep">
     <double>0.020000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="choice_x">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>260</y>
      <width>70</width>
      <height>40</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-200.000000000000000</double>
    </property>
    <property name="maximum">
     <double>200.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="choice_y">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>260</y>
      <width>70</width>
      <height>40</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-200.000000000000000</double>
    </property>
    <property name="maximum">
     <double>200.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="choice_z">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>260</y>
      <width>70</width>
      <height>40</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-200.000000000000000</double>
    </property>
    <property name="maximum">
     <double>200.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotate_x">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>370</y>
      <width>70</width>
      <height>40</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-180.000000000000000</double>
    </property>
    <property name="maximum">
     <double>180.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>5.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotate_y">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>370</y>
      <width>70</width>
      <height>40</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-180.000000000000000</double>
    </property>
    <property name="maximum">
     <double>180.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>5.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotate_z">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>370</y>
      <width>70</width>
      <height>40</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-180.000000000000000</double>
    </property>
    <property name="maximum">
     <double>180.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>5.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="red_value_2">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>560</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>255</string>
    </property>
   </widget>
   <widget class="QLabel" name="red_value_3">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>610</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>255</string>
    </property>
   </widget>
   <widget class="QLabel" name="red_value_4">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>660</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string>color: white;
font: 13pt &quot;Apple Braille&quot;;</string>
    </property>
    <property name="text">
     <string>255</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>glView</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">glview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
